src/main/java/org/veupathdb/eda/binaryfiles/dumper/IdFilesDumperFactory.java

```diff
@@ -0,0 +1,38 @@
+package org.veupathdb.eda.binaryfiles.dumper;
+
+import org.veupathdb.service.eda.ss.model.Entity;
+import org.veupathdb.service.eda.ss.model.Study;
+import org.veupathdb.service.eda.ss.model.variable.binary.BinaryFilesManager;
+
+import java.util.Map;
+
+public class IdFilesDumperFactory {
+  private BinaryFilesManager bfm;
+  private Study study;
+  private Entity entity;
+  private Entity parentEntity;
+
+  public IdFilesDumperFactory(BinaryFilesManager bfm,
+                              Study study,
+                              Entity entity,
+                              Entity parentEntity) {
+    this.bfm = bfm;
+    this.study = study;
+    this.entity = entity;
+    this.parentEntity = parentEntity;
+  }
+
+  /**
+   * Supplies an implementation of an ID FilesDumper based on the number of ancestors of entity member and the
+   * byte count allocated for IDs in ancestry.
+   */
+  public FilesDumper create(Map<String, Integer> bytesReservedForIdByEntityId) {
+    if (entity.getAncestorEntities().size() == 1) {
```

Comment:
Could return a switch here: case 0, case 1, default.  Not sure what would be clearer, but nice to not parse if conditions.

Reply:
Sure, I don't feel strongly but that's a bit more terse